docs/overview/creating_effects.md

```diff
@@ -23,7 +23,7 @@ You can also use methods in the companion objects of the `ZIO` type aliases:
 val s2: Task[Int] = Task.succeed(42)
 ```
 
-The `succeed` method is eager, which means the value passed to `succeed` will be constructed _before_ the method is invoked. Although this is the most common way to construct a successful effect, you can achieve lazy construction using the `ZIO.effectTotal` method:
+The `succeed` method takes a by-name parameter. It is the same as the `ZIO.effectTotal` method:
```

Comment:
How about something like?

"The `succeed` method takes a by-name parameter to make sure that any accidental side effects from constructing the value can be properly managed by the ZIO Runtime. However, `succeed` is intended for values which do not have any side effects. If you know that your value does have side effects consider using `ZIO.effectTotal` for clarity."

Reply:
That's a wonderful description that I didn't know before. 

A clarification: by-name parameter is not lazy -- it is like `def`, executed every time when it is used.

Reply:
However, the following sample code doesn't match the correct description -- there is no side effect in `mkString`. We should come with a better sample.

```scala mdoc:silent
lazy val bigList = (0 to 1000000).toList
lazy val bigString = bigList.map(_.toString).mkString("\n")

val s3 = ZIO.effectTotal(bigString)
```

Reply:
changed description and code sample in the following commit